components/style/values/specified/font.rs

```diff
@@ -1350,6 +1350,229 @@ impl Parse for FontVariantLigatures {
     }
 }
 
+bitflags! {
+    #[derive(MallocSizeOf)]
+    /// Vairants of numeric values
+    pub struct VariantNumeric: u8 {
+        /// None of other variants are enabled.
+        const NORMAL = 0;
+        /// Enables display of lining numerals.
+        const LINING_NUMS = 0x01;
+        /// Enables display of old-style numerals.
+        const OLDSTYLE_NUMS = 0x02;
+        /// Enables display of proportional numerals.
+        const PROPORTIONAL_NUMS = 0x04;
+        /// Enables display of tabular numerals.
+        const TABULAR_NUMS = 0x08;
+        /// Enables display of lining diagonal fractions.
+        const DIAGONAL_FRACTIONS = 0x10;
+        /// Enables display of lining stacked fractions.
+        const STACKED_FRACTIONS = 0x20;
+        /// Enables display of letter forms used with ordinal numbers.
+        const ORDINAL = 0x80;
+        /// Enables display of slashed zeros.
+        const SLASHED_ZERO = 0x40;
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl VariantNumeric {
+    /// Obtain a specified value from a Gecko keyword value
+    ///
+    /// Intended for use with presentation attributes, not style structs
+    pub fn from_gecko_keyword(kw: u8) -> Self {
+        Self::from_bits_truncate(kw)
+    }
+
+    /// Transform into gecko keyword
+    pub fn to_gecko_keyword(self) -> u8 {
+        self.bits()
+    }
+}
+
+impl ToCss for VariantNumeric {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        if self.is_empty() {
+            return dest.write_str("normal")
+        }
+
+        let mut has_any = false;
+
+        macro_rules! write_value {
+            ($ident:path => $str:expr) => {
+                if self.intersects($ident) {
+                    if has_any {
+                        dest.write_str(" ")?;
+                    }
+                    has_any = true;
+                    dest.write_str($str)?;
+                }
+            }
+        }
+
+        write_value!(VariantNumeric::LINING_NUMS => "lining-nums");
+        write_value!(VariantNumeric::OLDSTYLE_NUMS => "oldstyle-nums");
+        write_value!(VariantNumeric::PROPORTIONAL_NUMS => "proportional-nums");
+        write_value!(VariantNumeric::TABULAR_NUMS => "tabular-nums");
+        write_value!(VariantNumeric::DIAGONAL_FRACTIONS => "diagonal-fractions");
+        write_value!(VariantNumeric::STACKED_FRACTIONS => "stacked-fractions");
+        write_value!(VariantNumeric::SLASHED_ZERO => "slashed-zero");
+        write_value!(VariantNumeric::ORDINAL => "ordinal");
+
+        debug_assert!(has_any);
+        Ok(())
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl_gecko_keyword_conversions!(VariantNumeric, u8);
```

Comment:
@emilio Previously, in the PRs of `font-variant-east-asian` and `font-variant-ligatures`, I copied the content of `impl_gecko_keyword_conversions` macro. However, they will be looked like very duplicate codes.
Thus, if using macro is better, I'd like to send another PR to implement this macro for `font-variant-east-asian` and `font-variant-ligatures`.

Reply:
Yup, that sounds great!

Reply:
Great! Thanks; I'll send a PR for it!